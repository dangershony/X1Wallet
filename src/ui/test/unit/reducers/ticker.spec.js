import tickerReducer, { GET_TICKERS, RECIEVE_TICKERS } from 'reducers/ticker'

describe('reducers', () => {
  describe('tickerReducer', () => {
    it('should handle initial state', () => {
      expect(tickerReducer(undefined, {})).toMatchSnapshot()
    })

    it('should have GET_TICKER', () => {
      expect(GET_TICKERS).toEqual('GET_TICKERS')
    })

    it('should have RECIEVE_TICKER', () => {
      expect(RECIEVE_TICKERS).toEqual('RECIEVE_TICKERS')
    })

    it('should correctly getTicker', () => {
      expect(tickerReducer(undefined, { type: GET_TICKERS })).toMatchSnapshot()
    })

    it('should correctly receiveTicker', () => {
      expect(tickerReducer(undefined, { type: RECIEVE_TICKERS, ticker: 'foo' })).toMatchSnapshot()
    })
  })
})
