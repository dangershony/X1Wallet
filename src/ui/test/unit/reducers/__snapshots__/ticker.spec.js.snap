// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers tickerReducer should correctly getTicker 1`] = `
Object {
  "cryptoUnits": Object {
    "bitcoin": Array [
      Object {
        "key": "btc",
        "value": "BTC",
      },
      Object {
        "key": "bits",
        "value": "bits",
      },
      Object {
        "key": "sats",
        "value": "satoshis",
      },
    ],
    "litecoin": Array [
      Object {
        "key": "ltc",
        "value": "LTC",
      },
      Object {
        "key": "phots",
        "value": "photons",
      },
      Object {
        "key": "lits",
        "value": "litoshis",
      },
    ],
  },
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
    "UAH",
  ],
  "rates": null,
  "tickerLoading": true,
}
`;

exports[`reducers tickerReducer should correctly receiveTicker 1`] = `
Object {
  "cryptoUnits": Object {
    "bitcoin": Array [
      Object {
        "key": "btc",
        "value": "BTC",
      },
      Object {
        "key": "bits",
        "value": "bits",
      },
      Object {
        "key": "sats",
        "value": "satoshis",
      },
    ],
    "litecoin": Array [
      Object {
        "key": "ltc",
        "value": "LTC",
      },
      Object {
        "key": "phots",
        "value": "photons",
      },
      Object {
        "key": "lits",
        "value": "litoshis",
      },
    ],
  },
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
    "UAH",
  ],
  "rates": undefined,
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should handle initial state 1`] = `
Object {
  "cryptoUnits": Object {
    "bitcoin": Array [
      Object {
        "key": "btc",
        "value": "BTC",
      },
      Object {
        "key": "bits",
        "value": "bits",
      },
      Object {
        "key": "sats",
        "value": "satoshis",
      },
    ],
    "litecoin": Array [
      Object {
        "key": "ltc",
        "value": "LTC",
      },
      Object {
        "key": "phots",
        "value": "photons",
      },
      Object {
        "key": "lits",
        "value": "litoshis",
      },
    ],
  },
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
    "UAH",
  ],
  "rates": null,
  "tickerLoading": false,
}
`;
